### DISCUSSION TITLE
iOS: Changing the default map style

### DISCUSSION THREAD

**Discussion by r3econ** [maplibre/maplibre-native#2421] [Q&A]
I noticed that MapLibre starts loading and sets its default style before I apply my custom style. This results in map showing a default colorful MapLibre style before the correct style is fetched from the internet.

![image](https://github.com/maplibre/maplibre-native/assets/505054/0b85b107-7d9f-496c-b22f-25a4478e8ec4)
 
I can see the same behaviour when using a local JSON style file. The map attempts load of the default style before loading my custom style.

The style being loaded can be seen in `didFinishLoadingStyle` delegate call. There I found that the initial loaded style is:
```
 MLNStyle: name = "MapLibre", URL = "maplibre://maps/style
```
Default style load is followed by the load of my custom style.

Is there a way to overwrite the default MapLibre style so it does not load it whatsoever and goes straight into loading my custom style?

**Comment by louwers** [comment]
Can you share how you are setting your style?

**Comment by r3econ** [comment]
I'm instantiating the map in `makeUIView` and setting the style in `updateUIView` methods of the `UIViewRepresentable`:

```swift
struct MapViewRepresentable: UIViewRepresentable {

    func makeUIView(context: Context) -> MLNMapView {
        let mapView = MLNMapView()
        ...
        return mapView
    }
    
    func updateUIView(_ mapView: MLNMapView, context: Context) {
        mapView.styleURL = myRemoteStyleUrl
        ...
    }
}
```

I see in the delegate callbacks that the map is also trying to load the default style. In the calls below I'm logging style name:

![image](https://github.com/maplibre/maplibre-native/assets/505054/8cd69965-057b-479a-acba-f7e8302ca25c)

**Reply by louwers** [reply]
Can you try setting the `styleURL` in `makeUIView`? Or even pass it to the `MLNMapView` initializer?

**Comment by wipfli** [comment]
Looks like the behavior that was already reported here https://github.com/maplibre/maplibre-native/issues/709, right?
